### ISSUE TITLE
RViz crashes during MoveIt Quickstart in RViz

### ISSUE THREAD

**Issue by Jakeisthesnake** [ros-planning/moveit2_tutorials#883]
### Description

Overview of your issue here.
I am trying to run the MoveIt Quickstart in RViz tutorial. Unfortunately, RViz crashes during the launch.

### Your environment
* ROS Distro: Humble
* OS Version: Ubuntu 22.04
* Source build: Main

### Steps to reproduce
I have fairly clean build of Ubuntu 22.04. That said, building Moveit2 was giving me a hard time. I only built successfully once it build with:
`MAKEFLAGS="-j4 -l1" colcon build --executor sequential`

### Expected behavior
The demo should run as designed.

### Backtrace or Console output
ws_moveit$ ros2 launch moveit2_tutorials demo.launch.py
[INFO] [launch]: All log files can be found below /home/jake/.ros/log/2024-03-13-11-07-50-050628-jake-Blade-29447
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [rviz2-1]: process started with pid [29448]
[INFO] [static_transform_publisher-2]: process started with pid [29450]
[INFO] [robot_state_publisher-3]: process started with pid [29452]
[INFO] [move_group-4]: process started with pid [29454]
[INFO] [ros2_control_node-5]: process started with pid [29456]
[INFO] [spawner-6]: process started with pid [29458]
[INFO] [spawner-7]: process started with pid [29460]
[INFO] [spawner-8]: process started with pid [29462]
[rviz2-1] Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
[static_transform_publisher-2] [INFO] [1710353270.599045049] [static_transform_publisher]: Spinning until stopped - publishing transform
[static_transform_publisher-2] translation: ('0.000000', '0.000000', '0.000000')
[static_transform_publisher-2] rotation: ('0.000000', '0.000000', '0.000000', '1.000000')
[static_transform_publisher-2] from 'world' to 'base_link'
[robot_state_publisher-3] [INFO] [1710353270.626137892] [robot_state_publisher]: got segment base_link
[robot_state_publisher-3] [INFO] [1710353270.626282510] [robot_state_publisher]: got segment bracelet_link
[robot_state_publisher-3] [INFO] [1710353270.626302550] [robot_state_publisher]: got segment end_effector_link
[robot_state_publisher-3] [INFO] [1710353270.626314832] [robot_state_publisher]: got segment forearm_link
[robot_state_publisher-3] [INFO] [1710353270.626325727] [robot_state_publisher]: got segment half_arm_1_link
[robot_state_publisher-3] [INFO] [1710353270.626336031] [robot_state_publisher]: got segment half_arm_2_link
[robot_state_publisher-3] [INFO] [1710353270.626346514] [robot_state_publisher]: got segment robotiq_85_base_link
[robot_state_publisher-3] [INFO] [1710353270.626357257] [robot_state_publisher]: got segment robotiq_85_left_finger_link
[robot_state_publisher-3] [INFO] [1710353270.626367777] [robot_state_publisher]: got segment robotiq_85_left_finger_tip_link
[robot_state_publisher-3] [INFO] [1710353270.626378339] [robot_state_publisher]: got segment robotiq_85_left_inner_knuckle_link
[robot_state_publisher-3] [INFO] [1710353270.626388758] [robot_state_publisher]: got segment robotiq_85_left_knuckle_link
[robot_state_publisher-3] [INFO] [1710353270.626398671] [robot_state_publisher]: got segment robotiq_85_right_finger_link
[robot_state_publisher-3] [INFO] [1710353270.626408600] [robot_state_publisher]: got segment robotiq_85_right_finger_tip_link
[robot_state_publisher-3] [INFO] [1710353270.626418496] [robot_state_publisher]: got segment robotiq_85_right_inner_knuckle_link
[robot_state_publisher-3] [INFO] [1710353270.626443678] [robot_state_publisher]: got segment robotiq_85_right_knuckle_link
[robot_state_publisher-3] [INFO] [1710353270.626450662] [robot_state_publisher]: got segment shoulder_link
[robot_state_publisher-3] [INFO] [1710353270.626458236] [robot_state_publisher]: got segment spherical_wrist_1_link
[robot_state_publisher-3] [INFO] [1710353270.626465807] [robot_state_publisher]: got segment spherical_wrist_2_link
[robot_state_publisher-3] [INFO] [1710353270.626474813] [robot_state_publisher]: got segment world
[ros2_control_node-5] [INFO] [1710353270.638806992] [controller_manager]: Subscribing to '~/robot_description' topic for robot description file.
[ros2_control_node-5] [INFO] [1710353270.640367296] [controller_manager]: update rate is 1000 Hz
[ros2_control_node-5] [INFO] [1710353270.640584825] [controller_manager]: RT kernel is recommended for better performance
[ros2_control_node-5] [INFO] [1710353270.650818834] [controller_manager]: Received robot description file.
[ros2_control_node-5] [INFO] [1710353270.651640178] [resource_manager]: Loading hardware 'KortexMultiInterfaceHardware' 
[ros2_control_node-5] [INFO] [1710353270.652670593] [resource_manager]: Initialize hardware 'KortexMultiInterfaceHardware' 
[ros2_control_node-5] [WARN] [1710353270.652699997] [mock_generic_system]: Parameter 'fake_sensor_commands' has been deprecated from usage. Use'mock_sensor_commands' instead.
[ros2_control_node-5] [WARN] [1710353270.652734799] [mock_generic_system]: Parsing of optional initial interface values failed or uses a deprecated format. Add initial values for every state interface in the ros2_control.xacro. For example: 
[ros2_control_node-5] <state_interface name="velocity"> 
[ros2_control_node-5]   <param name="initial_value">0.0</param> 
[ros2_control_node-5] </state_interface>
[ros2_control_node-5] [INFO] [1710353270.652751403] [resource_manager]: Successful initialization of hardware 'KortexMultiInterfaceHardware'
[ros2_control_node-5] [INFO] [1710353270.652780335] [resource_manager]: Loading hardware 'RobotiqGripperHardwareInterface' 
[ros2_control_node-5] [INFO] [1710353270.652798744] [resource_manager]: Initialize hardware 'RobotiqGripperHardwareInterface' 
[ros2_control_node-5] [WARN] [1710353270.652813936] [mock_generic_system]: Parameter 'fake_sensor_commands' has been deprecated from usage. Use'mock_sensor_commands' instead.
[ros2_control_node-5] [INFO] [1710353270.652836302] [resource_manager]: Successful initialization of hardware 'RobotiqGripperHardwareInterface'
[ros2_control_node-5] [INFO] [1710353270.652904956] [resource_manager]: 'configure' hardware 'RobotiqGripperHardwareInterface' 
[ros2_control_node-5] [INFO] [1710353270.652926715] [resource_manager]: Successful 'configure' of hardware 'RobotiqGripperHardwareInterface'
[ros2_control_node-5] [INFO] [1710353270.652938272] [resource_manager]: 'activate' hardware 'RobotiqGripperHardwareInterface' 
[ros2_control_node-5] [INFO] [1710353270.652945529] [resource_manager]: Successful 'activate' of hardware 'RobotiqGripperHardwareInterface'
[ros2_control_node-5] [INFO] [1710353270.652953124] [resource_manager]: 'configure' hardware 'KortexMultiInterfaceHardware' 
[ros2_control_node-5] [INFO] [1710353270.652963177] [resource_manager]: Successful 'configure' of hardware 'KortexMultiInterfaceHardware'
[ros2_control_node-5] [INFO] [1710353270.652974837] [resource_manager]: 'activate' hardware 'KortexMultiInterfaceHardware' 
[ros2_control_node-5] [INFO] [1710353270.652981752] [resource_manager]: Successful 'activate' of hardware 'KortexMultiInterfaceHardware'
[move_group-4] [INFO] [1710353270.672341286] [move_group.moveit.RDFLoader]: Loaded robot model in 0.0052989 seconds
[move_group-4] [INFO] [1710353270.672404928] [move_group.moveit.robot_model]: Loading robot model 'gen3'...
[move_group-4] [INFO] [1710353270.672421428] [move_group.moveit.robot_model]: No root/virtual joint specified in SRDF. Assuming fixed joint
[ros2_control_node-5] [INFO] [1710353270.783256464] [controller_manager]: Received robot description file.
[ros2_control_node-5] [WARN] [1710353270.783301276] [controller_manager]: ResourceManager has already loaded an urdf file. Ignoring attempt to reload a robot description file.
[move_group-4] [INFO] [1710353270.832842673] [move_group.moveit.kdl_kinematics_plugin]: Joint weights for group 'manipulator': 1 1 1 1 1 1 1
[move_group-4] malloc(): invalid size (unsorted)
[ros2_control_node-5] [INFO] [1710353270.944747493] [controller_manager]: Loading controller 'joint_state_broadcaster'
[ros2_control_node-5] [INFO] [1710353270.957491361] [controller_manager]: Loading controller 'joint_trajectory_controller'
[ros2_control_node-5] [WARN] [1710353270.970609689] [joint_trajectory_controller]: [Deprecated]: "allow_nonzero_velocity_at_trajectory_end" is set to true. The default behavior will change to false.
[ros2_control_node-5] [INFO] [1710353270.974082403] [controller_manager]: Loading controller 'robotiq_gripper_controller'
[rviz2-1] [INFO] [1710353270.990812563] [rviz2]: Stereo is NOT SUPPORTED
[rviz2-1] [INFO] [1710353270.990990897] [rviz2]: OpenGl version: 4.6 (GLSL 4.6)
[spawner-6] [INFO] [1710353270.991016624] [spawner_joint_state_broadcaster]: Loaded joint_state_broadcaster
[ros2_control_node-5] [INFO] [1710353270.992116534] [controller_manager]: Configuring controller 'joint_state_broadcaster'
[ros2_control_node-5] [INFO] [1710353270.992237918] [joint_state_broadcaster]: 'joints' or 'interfaces' parameter is empty. All available state interfaces will be published
[spawner-7] [INFO] [1710353271.002768400] [spawner_joint_trajectory_controller]: Loaded joint_trajectory_controller
[spawner-6] [INFO] [1710353271.002828010] [spawner_joint_state_broadcaster]: Configured and activated joint_state_broadcaster
[ros2_control_node-5] [INFO] [1710353271.004004066] [controller_manager]: Configuring controller 'joint_trajectory_controller'
[ros2_control_node-5] [INFO] [1710353271.004223576] [joint_trajectory_controller]: No specific joint names are used for command interfaces. Using 'joints' parameter.
[ros2_control_node-5] [INFO] [1710353271.004273761] [joint_trajectory_controller]: Command interfaces are [position] and state interfaces are [position velocity].
[ros2_control_node-5] [INFO] [1710353271.004295607] [joint_trajectory_controller]: Using 'splines' interpolation method.
[ros2_control_node-5] [INFO] [1710353271.005583050] [joint_trajectory_controller]: Controller state will be published at 100.00 Hz.
[ros2_control_node-5] [INFO] [1710353271.007092888] [joint_trajectory_controller]: Action status changes will be monitored at 20.00 Hz.
[spawner-8] [INFO] [1710353271.011478647] [spawner_robotiq_gripper_controller]: Loaded robotiq_gripper_controller
[ros2_control_node-5] [INFO] [1710353271.013969993] [controller_manager]: Configuring controller 'robotiq_gripper_controller'
[ros2_control_node-5] [INFO] [1710353271.014043030] [robotiq_gripper_controller]: Action status changes will be monitored at 20Hz.
[rviz2-1] [INFO] [1710353271.015404363] [rviz2]: Stereo is NOT SUPPORTED
[spawner-7] [INFO] [1710353271.017632911] [spawner_joint_trajectory_controller]: Configured and activated joint_trajectory_controller
[spawner-8] [INFO] [1710353271.023057065] [spawner_robotiq_gripper_controller]: Configured and activated robotiq_gripper_controller
[rviz2-1] [WARN] [1710353271.139714520] [rcl.logging_rosout]: Publisher already registered for provided node name. If this is due to multiple nodes with the same name then all logs for that logger name will go out over the existing publisher. As soon as any node with that name is destructed it will unregister the publisher, preventing any further logs for that name from being published on the rosout topic.
[INFO] [spawner-6]: process has finished cleanly [pid 29458]
[rviz2-1] Warning: class_loader.impl: SEVERE WARNING!!! A namespace collision has occurred with plugin factory for class rviz_default_plugins::displays::InteractiveMarkerDisplay. New factory will OVERWRITE existing one. This situation occurs when libraries containing plugins are directly linked against an executable (the one running right now generating this message). Please separate plugins out into their own library or just don't link against the library and use either class_loader::ClassLoader/MultiLibraryClassLoader to open.
[rviz2-1]          at line 253 in /opt/ros/humble/include/class_loader/class_loader/class_loader_core.hpp
[INFO] [spawner-7]: process has finished cleanly [pid 29460]
[INFO] [spawner-8]: process has finished cleanly [pid 29462]
[rviz2-1] [INFO] [1710353271.176152493] [rviz2.RDFLoader]: Loaded robot model in 0.00404517 seconds
[rviz2-1] [INFO] [1710353271.176198601] [rviz2.robot_model]: Loading robot model 'gen3'...
[rviz2-1] [INFO] [1710353271.176211876] [rviz2.robot_model]: No root/virtual joint specified in SRDF. Assuming fixed joint
[rviz2-1] [ERROR] [1710353274.479695119] [rviz2.motion_planning_frame]: Action server: /recognize_objects not available
[rviz2-1] [INFO] [1710353274.512168557] [rviz2.motion_planning_frame]: MoveGroup namespace changed: / -> . Reloading params.
[rviz2-1] [INFO] [1710353274.569129278] [rviz2.RDFLoader]: Loaded robot model in 0.00339469 seconds
[rviz2-1] [INFO] [1710353274.569187604] [rviz2.robot_model]: Loading robot model 'gen3'...
[rviz2-1] [INFO] [1710353274.569195104] [rviz2.robot_model]: No root/virtual joint specified in SRDF. Assuming fixed joint
[rviz2-1] [INFO] [1710353274.688392692] [rviz2.kdl_kinematics_plugin]: Joint weights for group 'manipulator': 1 1 1 1 1 1 1
[rviz2-1] malloc(): invalid size (unsorted)
[ERROR] [rviz2-1]: process has died [pid 29448, exit code -6, cmd '/opt/ros/humble/lib/rviz2/rviz2 -d /home/jake/ws_moveit/install/moveit2_tutorials/share/moveit2_tutorials/launch/kinova_moveit_config_demo.rviz --ros-args -r __node:=rviz2 --params-file /tmp/launch_params_3gp4h4ru --params-file /tmp/launch_params_yf08k9sp --params-file /tmp/launch_params_rexkmkx2 --params-file /tmp/launch_params_gvv5k1nd --params-file /tmp/launch_params_50z88v97'].

[does not exit]

[In a separate terminal:]
~/ws_moveit$ ros2 topic list
/dynamic_joint_states
/joint_state_broadcaster/transition_event
/joint_states
/joint_trajectory_controller/controller_state
/joint_trajectory_controller/joint_trajectory
/joint_trajectory_controller/state
/joint_trajectory_controller/transition_event
/parameter_events
/robot_description
/robotiq_gripper_controller/transition_event
/rosout
/tf
/tf_static

~/ws_moveit$ ros2 node list
/controller_manager
/joint_state_broadcaster
/joint_trajectory_controller
/robot_state_publisher
/robotiq_gripper_controller
/static_transform_publisher

[re-opening rviz in another terminal]
~/ws_moveit$ ros2 run rviz2 rviz2 -d ws_moveit/src/moveit2_tutorials/doc/tutorials/quickstart_in_rviz/launch/kinova_moveit_config_demo.rviz
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
2024-03-13 11:49:25.219 [RTPS_TRANSPORT_SHM Error] Failed init_port fastrtps_port7419: open_and_lock_file failed -> Function open_port_internal
[INFO] [1710355765.428751431] [rviz2]: Stereo is NOT SUPPORTED
[INFO] [1710355765.428805425] [rviz2]: OpenGl version: 4.6 (GLSL 4.6)
[INFO] [1710355765.446369749] [rviz2]: Stereo is NOT SUPPORTED
Warning: class_loader.impl: SEVERE WARNING!!! A namespace collision has occurred with plugin factory for class rviz_default_plugins::displays::InteractiveMarkerDisplay. New factory will OVERWRITE existing one. This situation occurs when libraries containing plugins are directly linked against an executable (the one running right now generating this message). Please separate plugins out into their own library or just don't link against the library and use either class_loader::ClassLoader/MultiLibraryClassLoader to open.
         at line 253 in /opt/ros/humble/include/class_loader/class_loader/class_loader_core.hpp
[ERROR] [1710355780.595537032] [moveit_2102874026.motion_planning_frame]: Action server: /recognize_objects not available
[INFO] [1710355780.625788463] [moveit_2102874026.motion_planning_frame]: MoveGroup namespace changed: / -> . Reloading params.
[ERROR] [1710355790.676357208] [rviz]: Could not find parameter robot_description_semantic and did not receive robot_description_semantic via std_msgs::msg::String subscription within 10.000000 seconds.
Error:   Could not parse the SRDF XML File. Error=XML_ERROR_EMPTY_DOCUMENT ErrorID=13 (0xd) Line number=0
         at line 715 in ./src/model.cpp
[ERROR] [1710355790.692343543] [moveit_2102874026.RDFLoader]: Unable to parse SRDF
[ERROR] [1710355790.714138958] [moveit_2102874026.planning_scene_monitor]: Robot model not loaded

[does not exit]

Thanks for any advice!

**Comment by Jakeisthesnake** [comment]
My rqt graphs before and after I restarted RViz
![moveit_quickstart_fails_reopened_rviz](https://github.com/ros-planning/moveit2_tutorials/assets/28656653/64c2b87d-e583-49f3-a47f-9cf87eaa3c37)
![moveit_quickstart_fails](https://github.com/ros-planning/moveit2_tutorials/assets/28656653/d4531b53-ef33-47c6-b8fb-0abb8839c4ec)

**Comment by Jakeisthesnake** [comment]
Well, I never figured out exactly why I was getting a namespace issue. But I deleted my .cache, .ros, .rviz, and ws_moveit2 folders and went through the build process again and now things are working. I think maybe what happened what I tried to install Moveit following this page:
https://moveit.ros.org/install-moveit2/source/

and then went on to the getting started tutorial:
https://moveit.picknik.ai/main/doc/tutorials/quickstart_in_rviz/quickstart_in_rviz_tutorial.html

which basically has you reinstall moveit. That makes the most sense to me as far as why I might be getting a weird namespace issue. (I did end up deleting the ws_moveit2 folder before I even got to the namespace issue. Maybe just deleting the .cache folder would have been enough? Who knows.)
